src/main/java/org/candlepin/subscriptions/metering/service/prometheus/PrometheusMeteringController.java

```diff
@@ -76,65 +79,23 @@ public PrometheusMeteringController(ApplicationClock clock, PrometheusMetricsPro
     @Timed("rhsm-subscriptions.metering.openshift")
     @Transactional
     public void collectOpenshiftMetrics(String account, OffsetDateTime start, OffsetDateTime end) {
-        MetricProperties openshiftProperties = metricProperties.getOpenshift();
         openshiftRetry.execute(context -> {
             try {
-                log.info("Collecting OpenShift metrics");
-
-                QueryResult metricData = prometheusService.runRangeQuery(
-                    // Substitute the account number into the query. The query is expected to
-                    // contain %s for replacement.
-                    String.format(openshiftProperties.getMetricPromQL(), account),
-                    // Reset the start/end dates to ensure they span a complete hour.
-                    // NOTE: If the prometheus query step changes, we will need to adjust this.
-                    clock.startOfHour(start),
-                    clock.endOfHour(end),
-                    openshiftProperties.getStep(),
-                    openshiftProperties.getQueryTimeout()
-                );
-
-                if (StatusType.ERROR.equals(metricData.getStatus())) {
-                    throw new MeteringException(
-                        String.format("Unable to fetch OpenShift metrics: %s", metricData.getError())
-                    );
-                }
+                // Reset the start/end dates to ensure they span a complete hour.
+                // NOTE: If the prometheus query step changes, we will need to adjust this.
+                OffsetDateTime queryStart = clock.startOfHour(start);
+                OffsetDateTime queryEnd = clock.endOfHour(end);
```

Comment:
Is this something that we could use a Duration or Period class for?  Admittedly not familiar enough with the clock to know if this is a silly recommendation or not.

Reply:
If yes, there are also some unit tests that could do the same.